pyramid/tier_fs.go

```diff
@@ -232,19 +233,33 @@ func (tfs *TierFS) openFile(fileRef localFileRef, fh *os.File) (*ROFile, error)
 	}, nil
 }
 
-// readFromBlockStorage reads the referenced file from the block storage
+// openWithLock reads the referenced file from the block storage
 // and places it in the local FS for further reading.
 // It returns a file handle to the local file.
-func (tfs *TierFS) readFromBlockStorage(fileRef localFileRef) (*os.File, error) {
+func (tfs *TierFS) openWithLock(fileRef localFileRef) (*os.File, error) {
 	_, err := tfs.keyLock.Compute(fileRef.filename, func() (interface{}, error) {
-		var err error
+		// check again file existence, now that we have the lock
+		fh, err := os.Open(fileRef.fullPath)
+		if err == nil {
+			// file exists after all
+			cacheAccess.WithLabelValues(tfs.fsName, "Hit").Inc()
+			_ = fh.Close()
+			return nil, nil
```

Comment:
The file exists  and all is good; why not return it?
```suggestion
			return fh, nil
```

Reply:
In fact, I would expect this case to crash on the `tfs.openFile` call.

Reply:
See discussion below.